src/scenes/levelend-scene/levelend-scene.ts

```diff
@@ -142,6 +118,70 @@ export class LevelEndScene {
     }
   }
 
+  private createButton(
+    ButtonClass: typeof MapButton | typeof RetryButtonHtml | typeof NextButtonHtml,
+    id: string,
+    onClickCallback: () => void
+  ) {
+    const buttonsContainerId = 'buttons-container';
+
+    const button = new ButtonClass({ targetId: buttonsContainerId, id });
+    const gameControl = document.getElementById("game-control") as HTMLCanvasElement;
+    button.onClick(() => {
+      this.levelEndElement.style.zIndex = '7';
+      gameControl.style.zIndex = "-1";
+      onClickCallback();
+    });
+  };
+
+  renderButtonsHTML() {
+    // Define configurations for each button
+    const buttonConfigs = [
+      {
+        ButtonClass: MapButton,
+        id: 'levelend-map-btn',
+        onClick: () => {
+          this.handlePublishEvent(true);
+          this.switchToLevelSelectionCB();
+        },
+      },
+      {
+        ButtonClass: RetryButtonHtml,
+        id: 'levelend-retry-btn',
+        onClick: () => {
+          const gamePlayData = {
+            currentLevelData: {
+              ...this.data.levels[this.currentLevel],
+              levelNumber: this.currentLevel,
+            },
+            selectedLevelNumber: this.currentLevel,
+          };
+          this.handlePublishEvent(true, gamePlayData);
+          this.switchToGameplayCB();
+        },
+      },
+      {
+        ButtonClass: NextButtonHtml,
+        id: 'levelend-next-btn',
+        condition: this.isLastLevel,
+        onClick: () => {
+          const nextLevel = this.currentLevel + 1;
+          const gamePlayData = {
+            currentLevelData: { ...this.data.levels[nextLevel], levelNumber: nextLevel },
+            selectedLevelNumber: nextLevel,
+          };
+          this.handlePublishEvent(true, gamePlayData);
+          this.switchToGameplayCB();
+        },
+      },
+    ];
+  
+    // Create buttons based on configuration
+    buttonConfigs.forEach(({ ButtonClass, id, onClick, condition = true }) => {
+      if (condition) this.createButton(ButtonClass, id, onClick);
```

Comment:
For the IF condition here, it is not exactly clear what condition needed for the button to be created.

- Most of the object in the config array doesn't have a paremeter `condition,`, I this is confusing, you might want to add a default parameter so future developers would easily understand where to modify the condition if we want certain buttons on level end scene to be hidden.

Reply:
updated. pls check again @bernhardccodev